cockatrice/src/window_main.cpp

```diff
@@ -319,6 +319,9 @@ void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32
         case Response::RespClientIdRequired:
             QMessageBox::critical(this, tr("Error"), tr("This server requires client ID's. Your client is either failing to generate an ID or you are running a modified client.\nPlease close and reopen your client to try again."));
             break;
+        case Response::RespContextError:
+            QMessageBox::critical(this, tr("Error"), tr("An internal error has occurred, please try closing and reopening your client and try again. If the error persists try updating your client to the most recent build and if need be contact your software provider."));
```

Comment:
Like I mentioned, if its alright with you ill probably put a PR in to remove the check from the protocol handler portion of the code since there is a check on the server side that the operator can require the clientid.  With the protocol handler check in place it kind of makes the server side check useless.